tests/test_base.py

```diff
@@ -269,6 +269,28 @@ def test_exception_task(self, store_tracebacks):
         else:
             assert "traceback" not in execution
 
+    @pytest.mark.parametrize("max_stored_executions", [2, 3, 6, 11, None])
+    def test_max_stored_executions(self, max_stored_executions):
+        def _get_stored_executions():
+            return self.conn.llen(f"t:task:{task.id}:executions")
+
+        task = self.tiger.delay(
+            exception_task,
+            max_stored_executions=max_stored_executions,
+            retry_method=fixed(DELAY, 20),
+        )
+
+        assert _get_stored_executions() == 0
+
+        for __ in range(6):
+            Worker(self.tiger).run(once=True)
+            time.sleep(DELAY)
+
+            Worker(self.tiger).run(once=True)
+
+        assert task.n_executions() == 6
+        assert _get_stored_executions() == max_stored_executions or 6
```

Comment:
This assert is true, regardless of the values of `_get_stored_executions()` and `max_stored_executions`  (and also not what you intended even if it was parenthesized as `(max_stored_executions or 6)`).

I believe you want this:
```
assert _get_stored_executions() == min(max_stored_executions or 6, 6)
```

Reply:
I think it originally made sense, but the lack of parentheses prevented me from noticing when it ceased to make sense.

Fixed, thanks.